R/gitlab.R

```diff
@@ -0,0 +1,72 @@
+isGitlabURL <- function(url) {
+  is.string(url) && grepl("^http(?:s)?://(?:www|api).gitlab.(org|com)", url, perl = TRUE)
```

Comment:
Is it possible for users to host their own internal GitLab servers?

(looks like it is but that's soon going away: https://about.gitlab.com/gitlab-hosted/#githost-is-shutting-down-on-june-1st-2019; just want to double-check whether we need to support those)

Reply:
It looks like the thing that's going away is their single-tenant hosted server, but that you'll continue to be able to host your own GitLab server on-prem or in public cloud. Any thoughts on how to check for that?

It looks like for GitHub, we only permit github.com

```
isGitHubURL <- function(url) {
  is.string(url) && grepl("^http(?:s)?://(?:www|api).github.com", url, perl = TRUE)
}
```

Reply:
It looks like we skip the URL check in the main restore paths, e.g. for GitHub:

https://github.com/rstudio/packrat/blob/505f12aa4ef884cc7021719f4f0e4b69a0b93354/R/restore.R#L268

And you're doing the same for GitLab in the PR so I think this PR is fine as-is. Thanks!

Reply:
👍